### PR TITLE
Removing Empty Log Dirs

### PR DESCRIPTION
**Pull request by john-science**
## Description

This PR should help reduce the number of empty log directories after a run.

Essentially, it pulls the last vestigial pieces of the logging logic out of `context.py`. None of the logging is controlled at import time any more. (Which is good, we have far too much happening at import time.)

Instead, every time ARMI tries to log anything, it checks to see if the log dir is created, and if it's not, it creates it. And at the end of a simulation, where we concat log files together, we now hunt for empty log dirs and delete them.

This PR is risky. We have had several very specific requests as to how people wanted logging to work. This was 6-8 months ago, and people were very much concerned that logging works exactly as it does now. So this PR can not be merge into ARMI until we have had a chance to test the change with downstream projects, to see if this breaks people's workflows.

Obviously, I think this change was made with as light a hand as possible. But people spend a lot of time writing scripts to parse log files, so I can't guarantee this won't have unintended side effects for someone.

---

## Checklist

- [X] The code is understandable and maintainable to people beyond the author.
- [X] Tests have been added/updated to verify that the new or changed code works.
- [X] There is no commented out code in this PR.
- [X] The commit message follows [good practices](https://terrapower.github.io/armi/developer/tooling.html).
- [X] All docstrings are still up-to-date with these changes.

### CODE REVIEW

**Comment by john-science** [comment]
@joshuavictorchen  Just FYI. PR is up.

**Review by ntouran — APPROVED** [review]

**Review by joshuavictorchen — APPROVED** [review]

**Comment by john-science** [comment]
Okay, I'll be honest, I don't know why the coverage broke here.

I have been looking at it, but I can't reproduce the problem on my laptop.  It appears to be specific to whatever environmental change is on GitHub Actions.

This was working until I rolled the MPI-based tests into the code coverage. So, that's the big clue; the only clue I have to work with right now.

Sorry for the delay.

File: armi/nuclearDataIO/tests/library-file-generation/mc2v3-AB.inp [hunk 1]
```diff
@@ -8,8 +8,8 @@ $control
     l_gamma                 = T 
 /
 $library
-     c_mcclibdir  ="\\albert\apps\prod\mc2\3.2.2\libraries\endfb-vii.0\lib.mcc.e70"
-     c_gammalibdir = "\\albert\apps\prod\mc2\3.2.2\libraries\endfb-vii.0\lib.gamma.e70"
+     c_mcclibdir  ="\\path\to\mc2\3.2.2\libraries\endfb-vii.0\lib.mcc.e70"
+     c_gammalibdir = "\\path\to\mc2\3.2.2\libraries\endfb-vii.0\lib.gamma.e70"
```
Comment (jakehader) [inline armi/nuclearDataIO/tests/library-file-generation/mc2v3-AB.inp:12]: Oh interesting... I don't know what these inputs are needed for in the testing. They should probably not be in the framework.

Comment (john-science) [reply armi/nuclearDataIO/tests/library-file-generation/mc2v3-AB.inp]: Well, as you can see, these tests all run on my laptop with complete fake "\\path\to\" address.

I'm afraid I can't find documentation on whatever these ".inp" files are, so I don't know if we can just remove these lines entirely, and have the input files still be valid.

Do you know what these files are? Can these lines just be deleted?

**Pull request merged**